src/main/java/com/google/sps/servlets/ElectionServlet.java

```diff
@@ -59,24 +59,23 @@ public void doPut(HttpServletRequest request, HttpServletResponse response) thro
     String electionApiKey = ServletHelper.getApiKey("112408856470", "election-api-key", "1");
     URL url = new URL(BASE_URL + electionApiKey);
 
-    JSONObject obj = ServletHelper.readFromApiUrl(url, response);
+    JSONObject obj = ServletHelper.readFromApiUrl(url);
     JSONArray electionData = obj.getJSONArray("elections");
-    int numElections = electionData.length();
 
-    for (int i = 0; i < numElections; ++i) {
-      JSONObject currentElection = electionData.getJSONObject(i);
-      Entity electionEntity = new Entity("Election");
+    for (Object o : electionData) {
+      JSONObject election = (JSONObject) o;
 
       /* The "id" of an Election Entity is stored as a property instead of
        * replacing the Datastore-generated ID because  Datastore may
        * accidentally reassign IDs to other entities. To avoid this problem, I would have
        * to obtain a block of IDs with allocateIds(), but this is also difficult because
        * election IDs are not always consecutive numbers and other entities we plan to store
        * in Datastore will not have IDs from the Civic Information API (ex. policies) */
-      electionEntity.setProperty("id", currentElection.getLong("id"));
-      electionEntity.setProperty("name", currentElection.getString("name"));
-      electionEntity.setProperty("scope", currentElection.getString("ocdDivisionId"));
-      electionEntity.setProperty("date", currentElection.getString("electionDay"));
+      Entity electionEntity = new Entity("Election");
+      electionEntity.setProperty("electionId", election.getLong("id"));
```

Comment:
Can you change this to more object-oriented programming style?  Basically we want to delegate functions to objects.

For example, instead of declaring all of the code here, we can center all of the code around the AutoValue classes.

E.g. have each AutoValue implement these two methods: (using Election as an example):
```
public static Election fromJSONObject(JSonObject obj) {
  // creates a new Election object by extracting the properties from "obj"
} 

public Entity toEntity() {
  // creates a new Entity and sets the proper properties.
}
```

The pros is then you can hide all of the property names from the Servlet code, and reuse it across different servlets that also use that same entity from Civic Api and/or Datastore.

Reply:
Yeah, I'll look into this